experiment/simulation/js/main.js

```diff
@@ -1 +1,416 @@
+        opacity: '0.6'
+    });
+}
+
+function dropflask(){
+    const animation_1 = anime.timeline({
+        targets:'#conicalflask',
+        duration: 1000, 
+        easing: 'linear',
+    }).add({
+        translateY: '3.6vw',
+    });
+
+    const animation_2 = anime.timeline({
+        targets:'#water-flask',
+        duration: 800, 
+        easing: 'linear',
+    }).add({
+        translateY: '3.6vw',
+    });
+
+    const animation_3 = anime.timeline({
+        targets:'#bead',
+        duration: 1000, 
+        easing: 'linear',
+    }).add({
+        delay: '1000',
+        opacity: 1
+    }).add({
+        update: function(anim) {
+            message = messages[5];
+            document.getElementById("message").innerHTML = message;
+        }
+    });
+}
+
+function dropbead(){
+    const animation_1 = anime.timeline({
+        targets:'#bead',
+        duration: 1200, 
+        easing: 'linear',
+    }).add({
+        translateY: '7.6vw',
+    }).add({
+        update: function(anim) {
+            message = messages[6];
+            document.getElementById("message").innerHTML = message;
+        }
+    });
+}
+
+function stir_bead(){
+    const animation_1 = anime.timeline({
+        targets: "#bead", 
+        duration: 1000,
+        easing : 'linear',
+        loop: true,
+        direction: 'alternate',
+    });
+
+    animation_1.add({
+        width : '0.4vw',
+        rotateY: '10',
+        translateX: '0.4vw'
+    });
+
+    const animation_2 = anime.timeline({
+        targets: "#water-flask", 
+        duration: 3000,
+        easing : 'linear',
+    }).add({
+        update: function(anim) {
+            message = messages[8];
+            document.getElementById("message").innerHTML = message;
+        }
+    }).add({
+        delay: '3000',
+        update: function(anim) {
+            message = messages[9];
+            document.getElementById("message").innerHTML = message;
+        }
+    });
+
+}
+
+
+function heating() {
+    knob_heater_flag += 1;
+    if(knob_heater_flag === 2) {
+        const animation_2 = anime.timeline({
+            targets: "#pipette", 
+            duration: 500,
+            easing : 'linear',
+        }).add({
+            delay: '500',
+            opacity: 0.9,
+        }).add({
+            update: function(anim) {
+                message = messages[10];
+                document.getElementById("message").innerHTML = message;
+            }
+        });
+    }
+    else if(knob_heater_flag != 2)
+    {
+        const animation_1 = anime.timeline({
+            targets: '.message',
+            duration: 1000,
+            easing: 'linear'
+        }).add({
+            update: function(anim) {
+                message = messages[7];
+                document.getElementById("message").innerHTML = message;
+            }
+        });
+    }
+}
+
+function drop_pipette_to_beaker() {
+
+    console.log('pipette flag= ');
+    console.log(pipette_flag);
+    if(pipette_flag === 0)
```

Comment:
Can you use an enum to describe the different states of the pipette_flag instead of using numbers? enums are not supported natively in javascript but there are good alternatives available like using a map.

Reply:
Added comments and explanations